### PR TITLE
(docs) release notes for server 2.1.

### PR DESCRIPTION
**Pull request by jtappa**
based on initial notes from @dankreek. I left some as-is and filled in a few other descriptions.

### CODE REVIEW

File: documentation/release_notes.markdown [hunk 1]
```diff
@@ -23,45 +24,64 @@ Puppet Server 2.1 ships for the following supported platforms:
...
+* [SERVER-659](http://tickets.puppetlabs.com/browse/SERVER-659) - Restored broken http client timeout settings.
+* [SERVER-682](http://tickets.puppetlabs.com/browse/SERVER-682) - Fixed an issue where logback levels weren’t changed unless you restarted Puppet Server. This functionality had been provided in Puppet Server 1.0.2 but was inadvertently removed in Puppet Server 1.0.8, then merged with 2.0.
+* [SERVER-683](http://tickets.puppetlabs.com/browse/SERVER-683) - Changed the logic in the `legacy-routes-service` to get the route of the master-service via the service protocol name rather than by a hard-coded service name. This allows for the `legacy-routes-service` to pull the route from whatever service implementing the MasterService protocol happens to be in the service stack, i.e., master-service for open source Puppet or pe-master-service for PE.
+* [SERVER-684](http://tickets.puppetlabs.com/browse/SERVER-684) - Append `source_permissions=use` to 3.x file_metadata requests.
+* [SERVER-685](http://tickets.puppetlabs.com/browse/SERVER-685) - Investigated `environment_scenario` test failures.
```
Comment (camlow325) [inline documentation/release_notes.markdown]: I'd remove the note about SERVER-685.  Mostly a housekeeping ticket and doesn't affect the Puppet Server 2.1 release directly.

**Comment by camlow325** [comment]
:+1: @nfagerlund had raised the question about whether or not we should include history in these notes like we have been doing for the Puppet Server 1.x release notes - like whether the Puppet Server 2.0 notes should appear at the bottom of this document.  I'd personally be in favor of including that history here for consistency but will defer to the docs team's preferences.

**Comment by dankreek** [comment]
Looks great to me @jtappa ! :+1:

**Comment by dankreek** [comment]
@camlow325 shall we merge this, or are we waiting on a response from @nfagerlund ?

**Comment by camlow325** [comment]
@nfagerlund ?

**Pull request merged**